### PR TITLE
Fix findSubnet function's logic when subnet ID is specified

### PR DESCRIPTION
**Pull request by pydctw**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
findSubnet function's logic is incorrect when subnet ID is specified in AWSMachine's spec. The logic should find a subnet regardless of a failureDomain setting. Current logic returns a subnet ID from a spec when failureDomain is not defined, which may not match any subnets provided in AWSCluster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2732

**Special notes for your reviewer**:

**Checklist**:
- [ ] squashed commits
- [ ] includes documentation
- [x] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Fix findSubnet function's logic when subnet ID is specified. It will find a matching subnet regardless of a failureDomain setting.
```

### CODE REVIEW

**Comment by sedefsavas** [comment]
Nit. We usually create an issue before the PR especially if the issue is a bug, then open the PR referring that issue.

**Comment by pydctw** [comment]
> Nit. We usually create an issue before the PR especially if the issue is a bug, then open the PR referring that issue.

Thank you for the instruction. Didn't know that. Created an issue and linked to the PR description.

File: pkg/cloud/services/ec2/instances_test.go [hunk 1]
```diff
@@ -999,6 +999,163 @@ func TestCreateInstance(t *testing.T) {
...
+			expect: func(m *mock_ec2iface.MockEC2APIMockRecorder) {
+			},
+			check: func(instance *infrav1.Instance, err error) {
+				expectedErrMsg := "failed to run machine \"aws-test1\", subnet with id \"non-matching-subnet\" not found"
+				if err == nil {
+					t.Fatalf("Expected error, but got nil")
+				}
+
+				if !strings.Contains(err.Error(), expectedErrMsg) {
+					t.Fatalf("Expected error: %s\nInstead got: `%s", expectedErrMsg, err.Error())
```
Comment (shivi28) [inline pkg/cloud/services/ec2/instances_test.go:1155]: Minor Typo fix: It should be `Instead got: %s`

**Comment by pydctw** [comment]
@sedefsavas, could you approve the PR?

**Pull request merged**